packages/dev-server-core/src/server/DevServer.ts

```diff
@@ -19,6 +19,7 @@ export class DevServer {
     public config: DevServerCoreConfig,
     public logger: Logger,
     public fileWatcher = chokidar.watch([]),
+    externalServer?: Server,
```

Comment:
I think we have 2 optional params here now: `fileWatcher` and `externalServer`
so I'd make a new "options" argument, and try to support passing `fileWatcher` in an old manner for backwards compatibility and make `options.fileWatcher` and `options.externalServer` as new ones

Reply:
Yeah, I wondered if it made sense to have `externalServer` be part of `config`, being it's paired a bit with `middlewareMode`.

In that way, we could even make `externalServer` the truthy value of `middlewareMode`?

Reply:
Took: 
```
{
  middlewareMode: { server }
}
```

Reply:
more flexible indeed and still easy to use, great!